NEWS.md

```diff
@@ -147,6 +147,9 @@ Library improvements
     allowing `Diagonal` and `Bidiagonal` matrices with arbitrary
     `AbstractVector`s ([#22718], [#22925]).
 
+  * `Bidiagonal` constructors that automatically converted the input vectors to
+    the same type are deprecated in favor of explicit conversion ([#22925]).
```

Comment:
I was thinking this would go in the `Deprecated or removed` section since the library functionality part of the change is mentioned earlier